### DISCUSSION TITLE
Github Actions Support

### DISCUSSION THREAD

**Discussion by amelnychuck** [community/community#3034] [Mobile] [answered]
My team and I use GitHub actions to create releases, increment versions, and more for our iOS projects. We have flags to support manual triggers for these actions, and it would be incredible to have support for:
- Viewing actions and runs in the mobile app (useful for monitoring purposes)
- Trigger an action in the app (shortcuts support would be...incredible, even if there are some actions you reasonably might not want to do that for😂)
- Ability to view releases/draft releases in the app and finalize a draft release into a published one. Would love to do this workflow on my iPad

Thanks for all the wonderful work on the app and I'm already a very happy user even without these : )

**Comment by gleich** [comment]
This might be better as just regular status checks with some GitHub Actions specific stuff baked in. Right now the GitHub GraphQL API has status checks as part of their preview so I don't see them doing this soon. This would be great to have!

**Comment by zacwest** [comment]
I'd love to be able to view actions' results and tap a "notify me when this action completes" button to get a push notification (either desktop or mobile), since I'm generally not paying solid attention to the actions.

I'd love to trigger dispatch workflows in the app as well, but this is something I do less often on mobile than trying to view the status/results of an action.

**Comment by AaronMT** [comment]
I would love to have even a simple MVP of just viewing Workflow run status in the iOS app. I would probably not dispatch manually remotely but to just view status on my phone would be neat. I suppose I could setup notifications or use the site but am enjoying using the Github mobile iOS app more. I suppose on the iPad there's room for more detailed workflow information in landscape, but on my phone I just want status.

**Comment by finalchange23** [comment]
> My team and I use GitHub actions to create releases, increment versions, and more for our iOS projects. We have flags to support manual triggers for these actions, and it would be incredible to have support for:
> - Viewing actions and runs in the mobile app (useful for monitoring purposes)
> - Trigger an action in the app (shortcuts support would be...incredible, even if there are some actions you reasonably might not want to do that for😂)
> - Ability to view releases/draft releases in the app and finalize a draft release into a published one. Would love to do this workflow on my iPad
> 
> Thanks for all the wonderful work on the app and I'm already a very happy user even without these : )

**Comment by cpholguera** [comment]
I agree, at least being able to monitor the status and getting a notification whenever a workflow finishes it would be amazingly convenient! I imagine having that for our Release workflow, being notified of the auto-created "Release Draft", reviewing it and publishing it.. al straight from the Android/iOS app 💙💙

**Reply by Ibrahim579-opr** [reply]
> أوافق ، على الأقل أن تكون قادرًا على مراقبة الحالة والحصول على إشعار كلما انتهى سير العمل ، سيكون ذلك مناسبًا بشكل مثير للدهشة! أتخيل وجود ذلك لسير عمل الإصدار لدينا ، حيث يتم إخطارك بـ "مسودة الإصدار" التي تم إنشاؤها تلقائيًا ، ومراجعتها ونشرها .. مباشرة من تطبيق Android / iOS💙💙

**Comment by amrehman** [comment, marked as answer]
Hi there @amelnychuck ! thanks for this feature request and I agree with you.
I am going to add your note to our internal tracking issue and we will discuss within the team and will keep an eye out for similar user requests. This will really help us prioritise.

Thanks a lot and thank you for using our app :)

**Reply by alialhammad1992** [reply]
ok

**Reply by Hruuue663** [reply]
- [ ]

**Reply by The-Noah** [reply]
It appears to only be for PRs, and not the entire repo/specific commits? It would be awesome to get repo + commit level view of Actions. Still great work though!

**Comment by Angelk90** [comment]
@amrehman : There's news about that?

**Reply by amrehman** [reply]
Hi there @Angelk90 
Its at the moment in our backlog, stay tuned to our future releases :) We hope to give you an update soon!

**Reply by Angelk90** [reply]
Hi @amrehman , how long will it take before receiving the following feature, I don't know 4-5 months?

**Reply by JesseSkelly** [reply]
> Hi @amrehman , how long will it take before receiving the following feature, I don't know 4-5 months?

Yup

**Reply by peturssonstefan-ikea** [reply]
Hope this feature is still planned - super excited about it!

**Reply by Angelk90** [reply]
@amrehman : Besides the ability to see the actions that are in a project on mobile, would it be possible to add the ability to start a "Run workflow" action like in a browser?

**Comment by Racs663** [comment]
> My team and I use GitHub actions to create releases, increment versions, and more for our iOS projects. We have flags to support manual triggers for these actions, and it would be incredible to have support for:
> - Viewing actions and runs in the mobile app (useful for monitoring purposes)
> - Trigger an action in the app (shortcuts support would be...incredible, even if there are some actions you reasonably might not want to do that for😂)
> - Ability to view releases/draft releases in the app and finalize a draft release into a published one. Would love to do this workflow on my iPad
> 
> Thanks for all the wonderful work on the app and I'm already a very happy user even without these : )

- - [ ] [@oscarc@![image](https://user-images.githubusercontent.com/110435247/185768805-f1a36c0a-b584-4939-8020-11088996a5e5.jpeg)]()

**Comment by alialhammad1992** [comment]
ok

**Comment by Mehdimome** [comment]
Wi-fi

**Comment by Mehdimome** [comment]
Cod wi-fi

**Comment by Divinemoney** [comment]
Hey

**Comment by amrehman** [comment]
Hey folks!

We just shipped GitHub Actions on iOS and Android. 
https://github.blog/changelog/2022-10-04-introducing-actions-on-github-mobile/

Please give it a go and give us feedback :) Cheers!

**Reply by Angelk90** [reply]
@amrehman : The possibility of being able to execute a git action or to be able to see the status of the actions has not yet been implemented (I'm not talking about pr)?

**Reply by amrehman** [reply]
That is correct, @Angelk90 , we hope to address your use case in upcoming releases. Stay tuned !

**Reply by peturssonstefan-ikea** [reply]
Nice that it's now available for  the PR's.
I'd like to second @Angelk90 request. Thought this feature would allow us to gain an overview of all the actions. In particular I'm missing access to view the scheduled jobs too. 

That said, thanks for all the good work!

**Reply by amrehman** [reply]
Absolutely! Thanks a lot for this awesome feedback! Cheers

**Reply by hfhbd** [reply]
How can you see the status list of a repository on mobile, like the action tab on web?

**Reply by Wladefant** [reply]
@amelnychuck @amrehman I think this should be market as th3 answer

**Reply by ka592** [reply]
> @amelnychuck @amrehman I think this should be market as th3 answer

**Reply by ka592** [reply]
> @amrehman : Is there any news?

**Reply by ka592** [reply]
> @amrehman : Is there any news?

**Reply by ka592** [reply]
> How can you see the status list of a repository on mobile, like the action tab on web?

**Reply by ka592** [reply]
> Absolutely! Thanks a lot for this awesome feedback! Cheers

**Reply by Wladefant** [reply]
@amrehman this discussion is now resolved and can be closed

**Reply by Angelk90** [reply]
@Wladefant : Solved what, I don't see anything solved?

**Reply by fractos** [reply]
Android mobile version does not show all workflows - the list cuts off after x many (50?), and (obviously) the ones I want to look at are beyond this limit. Frustrating.

**Comment by ijuuu930104** [comment]
Github Actions Support

**Reply by ka592** [reply]
> Github Actions Support

**Comment by AilMilk1100** [comment]
@

**Reply by ka592** [reply]
> @

**Comment by pelikhan** [comment]
So we can browse action results for PR but not commits, very frustrating.

**Reply by ka592** [reply]
> So we can browse action results for PR but not commits, very frustrating.

**Comment by sim-cha** [comment]
> My team and I use GitHub actions to create releases, increment versions, and more for our iOS projects. We have flags to support manual triggers for these actions, and it would be incredible to have support for:
> 
> * Viewing actions and runs in the mobile app (useful for monitoring purposes)
> * Trigger an action in the app (shortcuts support would be...incredible, even if there are some actions you reasonably might not want to do that for😂)
> * Ability to view releases/draft releases in the app and finalize a draft release into a published one. Would love to do this workflow on my iPad
> 
> Thanks for all the wonderful work on the app and I'm already a very happy user even without these : )

I am looking for a direct way to link/trigger the "Run Workflow" button
Any help would be greatly appreciated .
<img width="363" alt="Run workflow " src="https://github.com/community/community/assets/6885309/4dde842b-cc75-4910-b001-84c2a2a573ce">

**Comment by wzhiana** [comment]
as this is the most voted discussion and the oldest, i'll add all related requests:
* https://github.com/orgs/community/discussions/70332
* https://github.com/orgs/community/discussions/74694
* https://github.com/orgs/community/discussions/70159
* https://github.com/orgs/community/discussions/58253
* https://github.com/orgs/community/discussions/51159

so strange, that gh still not created easy way to generate links to trigger actions directly from simple html. Will be extremely usefull to add some links in releases, like: 
deploy to: [prod]:http://gh/actions/some_deploy.yml?where=to_prod
and if there is need to do some verification or whatever, gh can show pop-up page: "are you sure?" and by cliking yes - send post request. 

but it will make life so easier for people who not develop, but forced to use GH due to corporate policies.

**Reply by wzhiana** [reply]
@amrehman if this still not realized, may i ask to add some links from discussions to add more weight for that feature?

**Comment by abosaod183** [comment]
I agree, at least being able to monitor the status and getting a notification whenever a workflow finishes it would be amazingly convenient! I imagine having that for our Release workflow, being notified of the auto-created "Release Draft", reviewing it and publishing it.. al straight from the Android/iOS app 💙💙
